pkgs/by-name/ca/castero/package.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, fetchFromGitHub
+, python3
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "castero";
+  version = "0.9.5";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "xgi";
+    repo = "castero";
+    rev = "v${version}";
+    hash = "sha256-6/7oCKBMEcQeJ8PaFP15Xef9sQRYCpigtzINv2M6GUY=";
+  };
+
+  nativeBuildInputs = with python3.pkgs; [
+    setuptools
+    wheel
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    requests
+    grequests
+    cjkwrap
+    pytz
+    beautifulsoup4
+    lxml
+    mpv
+    python-vlc
+  ];
+
+  nativeCheckInputs = with python3.pkgs; [
+    pytest
+    coverage
+  ];
+
+  pytestFlagsArray = [
+    "tests"
+  ];
+
+  pythonImportCheck = [
+    "castero"
+  ];
+
+  # Satisfy the python-mpv depedency, which is mpv within NixOS
+  postPatch = ''
+    substituteInPlace setup.py --replace-fail "python-mpv" "mpv"
+  '';
+
+  meta = with lib; {
+    mainProgram = "castero";
+    description = "TUI podcast client for the terminal";
+    homepage = "https://github.com/xgi/castero";
+    license = licenses.mit;
+    maintainers = with maintainers; [ keto ];
```

Comment:
maybe we should set platforms from vlc ? currently vlc doesnt support darwin, but this derivation uses platforms from python

logs
https://logs.ofborg.org/?key=nixos/nixpkgs.303771&attempt_id=4cc4b9b3-7f62-448b-8031-fe95cf101c1a

Reply:
I'm also trying to run tests (since the repository has them), how do I go about doing that? `pytestFlagsArray` doesn't seem to do the trick

Reply:
add pytestCheckHook to nativeCheckInputs